### PR TITLE
AllowEditPost and PostEditTimeLimit migration

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

**Comment by jespino** [comment]
One think is the permission, and other thing is the time limit limitation, here: https://github.com/mattermost/mattermost-server/pull/8208/files#diff-317b12c12f71e311f13813107f4d8550R351 we check if you have a license and then, if the timelimit is set to something different from -1, and finally if the time of the post if out of the expected range. Then, return an error. It happens for each UpdatePost and PatchPost (which use UpdatePost).

**Comment by mkraft** [comment]
I'm not 100% understanding how `ALLOW_EDIT_POST_NEVER` is being handled. Shouldn't `system_admin` not have `edit_post` permimssion in some cases?

**Comment by jespino** [comment]
You are right, when the ALLOW_EDIT_POST is NEVER the system admin can't edit too. So, I have to remove the PERMISSION_EDIT_POST from global system admin permissions and add based on ALLOW_EDIT_POST config, give me a minute to change it.

**Comment by jespino** [comment]
Done

**Comment by grundleborg** [comment]
Looks like we need to merge master into advanced-permissions-phase-1 again and rebase this branch, to get the tests to pass. There have been enterprise changes.

**Comment by jespino** [comment]
Ok, who do it? I can do it.

**Comment by grundleborg** [comment]
@jespino go for it :)

**Comment by jespino** [comment]
Merged master into advanced-permissions-phase-1, and rebased branch

File: utils/authorization.go [hunk 1]
```diff
@@ -267,5 +267,37 @@ func SetRolePermissionsFromConfig(roles map[string]*model.Role, cfg *model.Confi
 		)
 	}
 
+	if IsLicensed() {
+		switch *cfg.ServiceSettings.AllowEditPost {
+		case model.ALLOW_EDIT_POST_ALWAYS:
```
[the remark this answers is not in the document]
Comment (jespino) [reply utils/authorization.go]: Sure, the problem was I'm a bit newbie to golang yet.

[the remark this answers is not in the document]
Comment (mkraft) [reply utils/authorization.go]: Yeah me too, I had no idea it was possible until I looked it up.

[the remark this answers is not in the document]
Comment (jespino) [reply utils/authorization.go]: Done

**Pull request merged**